### PR TITLE
[14.0][FIX] stock_reserve_rule: fix _apply_strategy_empty_bin

### PR DESCRIPTION
**Pull request by mt-software-de**
Empty bin Rule
Now checks if there is an other product which is allocating the location

### CODE REVIEW

**Comment by mt-software-de** [comment]
@jbaudoux could please have a look?

File: stock_reserve_rule/models/stock_reserve_rule.py [hunk 1]
```diff
@@ -222,8 +222,18 @@ def _apply_strategy_empty_bin(self, quants):
 
         # We take goods only if we empty the bin.
         # The original ordering (fefo, fifo, ...) must be kept.
-        rounding = fields.first(quants).product_id.uom_id.rounding
+        product = fields.first(quants).product_id
+        rounding = product.uom_id.rounding
         for location, location_quants in quants_per_bin:
+            if self.env["stock.quant"].search(
```
Comment (jbaudoux) [inline stock_reserve_rule/models/stock_reserve_rule.py]: Can't we lower the amount of queries ?

**Comment by jbaudoux** [comment]
cc @sebalix

File: stock_reserve_rule/models/stock_reserve_rule.py (hunk 1 above)
Comment (jbaudoux) [reply stock_reserve_rule/models/stock_reserve_rule.py]: use read_group instead outside of for loop

**Review by jbaudoux — APPROVED** [review]

**Review by sebalix — APPROVED** [review]
LG

**Pull request merged**